### PR TITLE
Unconditionally style contents of `L<>` tags

### PR DESCRIPTION
**Pull request by mjsir911**
This lets users properly highlight man(1) references even if guesswork doesn't report it as a manref. The offending links looked something like this:

`@@RXVT_NAME@@perl(3)`

Being able to wrap these in `L<>` to bold the entire string is a nicety.

A (too) thorough writeup, exploration, & justification of this PR can be found [here](https://sirabella.org/blog/rabbitholes)

TODO:
- [ ] fix tests
- [ ] add another test outlining specific functionality
- [ ] find a better way to apply bold than manual format codes?

### CODE REVIEW

File: lib/Pod/Man.pm [hunk 1]
```diff
@@ -1321,8 +1323,11 @@ sub cmd_l {
         } else {
             return "$text <$$attrs{to}>";
         }
+    } elsif ($$attrs{type} eq 'man') {
+        my ($page, $section) = $text =~ /^([^(]+)(?:[(](\d+)[)])?$/; # Copied from Simple.pm
+        return '\f(BS' . $page . '\f(BE\|' . "($section)";
```
Comment (mjsir911) [inline lib/Pod/Man.pm:1328]: Is there a better way to apply the bold formatting than using the format codes?

**Comment by rra** [comment]
Thank you for the analysis and the initial patch! I have a committed a fix for this that will be in the next release.

The problem was a bit more complex than your patch, because by the time `cmd_l` gets the text, it has already been formatted, including man page references. To ensure the formatting is correct, `cmd_l` needs to use the underlying link attributes and reproduce the formatting. Otherwise, it risks double-formatting the reference in some edge cases or, more likely, missing some formatting. I think I found a way to do this that should work reliably, and added a test for it.

**Comment by mjsir911** [comment]
Thank you @rra
